test/new-e2e/tests/windows/common/network.go

```diff
@@ -80,8 +82,27 @@ func ListBoundPorts(host *components.RemoteHost) ([]*BoundPort, error) {
 // If the URL is a local file, it will be uploaded to the VM.
 // If the URL is a remote file, it will be downloaded from the VM
 func PutOrDownloadFile(host *components.RemoteHost, url string, destination string) error {
+	// no retry
+	return PutOrDownloadFileWithRetry(host, url, destination, &backoff.StopBackOff{})
+}
+
+// PutOrDownloadFileWithRetry is similar to PutOrDownloadFile but retries on download failure,
+// local file copy is not retried.
+func PutOrDownloadFileWithRetry(host *components.RemoteHost, url string, destination string, b backoff.BackOff) error {
 	if strings.HasPrefix(url, "http://") || strings.HasPrefix(url, "https://") {
-		return DownloadFile(host, url, destination)
+		err := backoff.Retry(func() error {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Would it be possible to pass a `testing.T` context in the signature, so that we could use `testify.Eventually` instead of `backoff`? We had plans to remove `backoff` from `test/new-e2e` as it is all code uses in tests, makes sense when possible to use testing features.

Reply:
I'm trying to avoid `testify.Eventually` in code that isn't specifically a test due to the issues we've encountered with it. Have those been resolved, yet?

Reply:
Which issue did you encounter?

Reply:
`testify.Eventually` waits for the delay interval once before calling the condition function.

Reply:
[The PR that checks](https://github.com/stretchr/testify/issues/1439) the condition before running the delayed loop is still open. Is this a blocker in this case?

Reply:
the download succeeding the first time is the 99% case yes.